lib/brcobranca/calculo.rb

```diff
@@ -22,91 +22,21 @@ def modulo10
       valor == 10 ? 0 : valor
     end
 
-    # Calcula módulo 11 com multiplicaroes de 9 a 2 segundo a BACEN.
+    # Calcula o módulo 11 segundo a BACEN
     #
     # @return [Integer]
-    def modulo11_9to2
-      total = multiplicador([9, 8, 7, 6, 5, 4, 3, 2])
-
-      (total % 11)
-    end
-
-    # Calcula módulo 11 com multiplicaroes de 2 a 9 segundo a BACEN.
-    #
-    # @return [Integer]
-    def modulo11_2to9
-      total = multiplicador([2, 3, 4, 5, 6, 7, 8, 9])
-
-      valor = (11 - (total % 11))
-      [0, 10, 11].include?(valor) ? 1 : valor
-    end
-
-    # Calcula módulo 11 com multiplicadores de 2 a 9 e 2 a 5 (Utilizado pelo Santander).
-    #
-    # @return [Integer]
-    def modulo11_santander
-      return 0 if to_i == 0
-
-      somatorio = 0
-      multiplicadores = [2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5]
-      base = "#{rjust(12, '0')}".reverse
-      index = 0
-
-      base.each_char do |_char|
-        somatorio += base[index].to_i * multiplicadores[index].to_i
-        index += 1
-      end
-
-      resto = somatorio % 11
-      return 1 if resto == 10
-      return 0 if resto == 1 || resto == 0
-      11 - resto
-    end
-
-    # Calcula módulo 11 com multiplicadores de 2, 7 a 2, e 7 a 2 (Utilizado pelo Bradesco).
-    #
-    # @return [Integer, String] Caso o resto seja 1, retorna P
-
-    def modulo11_bradesco
-      somatorio = 0
-      multiplicadores = [2, 7, 6, 5, 4, 3, 2, 7, 6, 5, 4, 3, 2]
-      index = 0
-
-      each_char do |_char|
-        somatorio += self[index].to_i * multiplicadores[index]
-        index += 1
-      end
-
-      resto = somatorio % 11
-      return 0 if resto == 0
-      return 'P' if resto == 1
-      11 - resto
-    end
-
-    # Calcula módulo 11 com multiplicaroes de 2 a 9 (Utilizado pela CAIXA - boletos SIGCB).
-    #
-    # @return [Integer]
-    def modulo11_2to9_caixa
-      total = multiplicador([2, 3, 4, 5, 6, 7, 8, 9])
-      total = (total % 11) unless total < 11
-      valor = (11 - total)
-      valor > 9 ? 0 : valor
-    end
+    # @raise  [ArgumentError] Caso não seja um número inteiro.
+    # @param  [Hash] options Opções para o cálculo do módulo
+    # @option options [Hash] :mapeamento Mapeamento do valor final. Ex: { 10 => "X" }. Padrão: {}
+    # @option options [Array] :multiplicador Números a serem utilizados na multiplicação da direita para a esquerda. Padrão: [9 até 2]
+    def modulo11(options={}, &block )
```

Comment:
Unused method argument - `block`. If it's necessary, use `_` or `_block` as an argument name to indicate that it won't be used.<br>Surrounding space missing in default value assignment.<br>Space inside parentheses detected.